tests/integration_tests/functional/test_rate_limiter.py

```diff
@@ -158,95 +157,50 @@ def _check_tx_rate_limiting(test_microvm, guest_ips, host_ips):
     _start_local_iperf(test_microvm.jailer.netns_cmd_prefix())
 
     # First step: get the transfer rate when no rate limiting is enabled.
-    # We are receiving the result in KBytes from iperf; 1000 converts to Bytes.
-    iperf_cmd = '{} -c {} -t{} -f KBytes -O {}'.format(
+    # We are receiving the result in KBytes from iperf.
+    iperf_cmd = '{} -c {} -t {} -f KBytes -w {} -N'.format(
```

Comment:
I think would be better to test with UDP, TCP has control packets and retransmissions that might go over the set threshold.

Reply:
Like I said, moving to UDP involves some effort to calculate total wire-bytes vs total protocol bytes (iperf only reports protocol bw). This difference is much bigger with UDP vs TCP in iperf3 because UDP packet size is hardcoded to a small value.

If we want extra effort invested here, we should just switch to a different tool that can be as simple as send a fixed amount of bytes and measure time.

Reply:
I don't get this. On UDP you have no protocol options and you always know the header size and iperf allows you to specify the packet/payload size. You can also use the counters on the tap interface to check the number of bytes.

Reply:
I've tested with UDP. I couldn't find a combination of packet-size/bandwidth that provides smoother traffic than current TCP test..

The biggest problem (for short-duration tests) remains the fact that our rate-limiter timer resolution is a coarse `100ms`. This translates to 100ms cut-off periods, which are very big cut-off periods for tests of 2-3 seconds (statistically skewing the end result).

I say we go with the version of the test in this PR which should work well. Most runs will pass with their short version. For the ones that fail the short test (statistically bound to happen with unfortunate timings between device cut-offs and iperf measurements), a longer one is automatically run and that one determines the result.
This strategy keeps runs short in most cases, the outliers will take more time, but the test itself is reliable and doesn't provide false positives or negatives.

**Note**: average test duration should be lower 👍 since I've removed traffic-omit periods.